src/lib/mina_lib/mina_lib.ml

```diff
@@ -1033,6 +1033,65 @@ let perform_compaction t =
       in
       perform interval_configured
 
+let daemon_start_time = Time_ns.now ()
+
+let check_and_stop_daemon t =
+  let interval_mins = (40 * 60) + (Random.int 16 * 30) in
```

Comment:
Computing the random delta inside of this function means we will keep recomputing it each time we loop back here. I think this should just be computed once outside instead (don't need to worry about it if you take my suggestion on line 1093 though)

Reply:
I added this within the loop for extra variability

Reply:
Wouldn't this just end up skewing everyone to exit at early times in the interval? If you run this function every 3 mins and generate a new random number for the threshold, you will end up with a greater chance to exit in an earlier portion of that interval, so it's basically a weighted random distribution. Computing this once up front would give a truly random distribution (well, as random as `Random.int` can produce, at least), so that would give you greater variability than computing this within the loop for each threshold check.

Reply:
Hmm, yeah, I should also probably make it an hour apart- `(Random.int 8 * 60)`